### PR TITLE
3D beta rendering support for SceneEntity `texts`

### LINKED ISSUE #4319: 3D beta rendering support for SceneEntity `texts`

### PR DESCRIPTION
**Pull request by snosenzo**
**User-Facing Changes**
 - Allow for the rendering of the Text scene entity primitive

**Description**
 - added textprimitive class to scene entities using the LabelPool from the renderer
  - updated scene entity story to include text example

Fixes #4319

### CODE REVIEW

**Comment by jtbandes** [comment]
For the story, could you include examples of billboard=false as well? And maybe make the large one a bit smaller so it leaves room for us to add more to the story :)

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        text.pose.orientation.w,
+      );
+
+      label.position.set(text.pose.position.x, text.pose.position.y, text.pose.position.z);
+      label.userData.pose = text.pose;
+      i++;
+    }
+    // need to release the no longer used labels so that they don't linger on the scene
+    if (i < this.labels.length) {
+      let j = this.labels.length;
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts:89]: Probably simpler to change the above loop to `for (; i < texts.length; i++)` and then this loop can be `for (; i < labels.length; i++)`. Or even simpler, `for (const label of this.labels.splice(i)) labelPool.release(label)`

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    if (newLength > oldLength) {
+      for (let i = oldLength; i < newLength; i++) {
+        const newLabel = this.labelPool.acquire();
+        this.labels.push(newLabel);
+        this.add(newLabel);
+      }
+    }
+  }
+
+  private _updateMesh(texts: TextPrimitive[]) {
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts:45]:
```suggestion
  private _updateTexts(texts: TextPrimitive[]) {
```

I regret naming this method something different in all of the renderables 🤣

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      label.setSizeAttenuation(!text.scale_invariant);
+      label.quaternion.set(
+        text.pose.orientation.x,
+        text.pose.orientation.y,
+        text.pose.orientation.z,
+        text.pose.orientation.w,
+      );
+
+      label.position.set(text.pose.position.x, text.pose.position.y, text.pose.position.z);
+      label.userData.pose = text.pose;
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts:84]: Is this line needed? I think it's taken care of by setting label.position & label.quaternion

Comment (snosenzo) [reply packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts:84]: isn't this used for selection? (though I know selecting single instances of entities doesn't work yet)

**Review by jtbandes — APPROVED** [review]

**Pull request merged**